Wrote section in database-processes about how database processes are … (#267)
Starcounter/Docs: docs/database-processes.md

Unhandled exception. System.Exception: ScErrCantStartDatabase (SCERR10004): Attempting to start the database failed. Failed to start ConsoleApp. Data process reported ScErrDbAlreadyStarted (SCERR2108): A Starcounter database with the specified name is already started.
```

> Starcounter tooling or management components - and not end-users - normally control `scdata` processes and their lifetime.

Reviewer: ```suggestion
Starcounter tooling or management components control `scdata` processes and their lifetime. Under normal circumstances no manual interaction with the `scdata` process shall be required from Starcounter end users.
```

> Starcounter tooling or management components control `scdata` processes and their lifetime. Under normal circumstances no manual interaction with the `scdata` process shall be required from Starcounter end users.

Reviewer 2: ```suggestion
Starcounter tooling or management components control `scdata` processes and their lifetimes. Under normal circumstances no manual interaction with the `scdata` process shall be required from Starcounter end users.
```

### Deciding if a database process is started

> When a request come to start a database, Starcounter need to make sure corresponding database process is running and, if not, start one. A client, such as an application asking to connect to Starcounter, specifies the path to the database directory, and Starcounter combines this with an internal, opaque version number to determine if the process is running.

Reviewer: ```suggestion
When a request to start a database comes, Starcounter starts corresponding database process if it's not running. A client, such as an application asking to connect to Starcounter, specifies the path to the database directory, and Starcounter combines this with an internal, opaque database engine version number to determine if the process is running.
```

Let's look at two examples:

> Application connecting to Starcounter:

Reviewer: ```suggestion
An application connects to Starcounter:
```

> When a request to start a database comes, Starcounter starts corresponding database process if it's not running. A client, such as an application asking to connect to Starcounter, specifies the path to the database directory, and Starcounter combines this with an internal, opaque database engine version number to determine if the process is running.

Reviewer 2: ```suggestion
When receiving a request to to start a database, Starcounter starts the corresponding database process if it's not already running. A client, such as an application asking to connect to Starcounter, specifies the path to the database directory, and Starcounter combines this with an internal, opaque database engine version number to determine if the process is running.
```

> * Application ask to connect to `./mydb`, i.e `Database=./mydb`.

Reviewer: ```suggestion
* Application asks to connect to `./mydb`, i.e `Database=./mydb`.
```

> * Application asks to connect to `./mydb`, i.e `Database=./mydb`.

Reviewer 2: ```suggestion
* During startup, the application requests a connection to the database `mydb`, located in the root folder. This is indicated by the `Database=./mydb` segment in the connection string.
```

* Starcounter checks the internal version of Starcounter binaries referenced by the **application**, e.g. "3.2.1".

> * Starcounter combine the two above values to form a unique database identity, and use that to check if the database process is running.

Reviewer: ```suggestion
* Starcounter combines the two above values to form a unique database identity, and use that to check if the database process is running.
```

> Tooling used to start an application:

Reviewer: ```suggestion
Starcounter tool starts a database:
```

> * Starcounter combines the two above values to form a unique database identity, and use that to check if the database process is running.

Reviewer 2: ```suggestion
* Starcounter combines the two above values to form a unique database identity, and uses that to check if the database process is running.
```

> * Tooling is used to start the database i.e. `dotnet start start ./mydb`.

Reviewer 2: ```suggestion
* The Starcounter command line interface (CLI) is used to start the database `mydb`. This can be done with the command `dotnet star start ./mydb`.
```

* Starcounter checks the internal version of Starcounter binaries referenced by the **tool**, e.g. "3.2.1".

> * Starcounter combine the two above values to form a unique database identity, and use that to check if the database process is running.

Reviewer: ```suggestion
* Starcounter combines the two above values to form a unique database identity, and use that to check if the database process is running.
```

> * Starcounter combines the two above values to form a unique database identity, and use that to check if the database process is running.

Reviewer 2: ```suggestion
* Starcounter combines the two above values to form a unique database identity, and uses that to check if the database process is running.
```

> If two requests come to start the same database (same path), but _versions mismatch_, the second request will fail, saying that the database is already started (but with another, incompatible version of Starcounter binaries). 

Reviewer 2: ```suggestion
If two requests come to start the same database (same path), but _versions mismatch_, the second request will fail, saying that the database is already started (but with another, incompatible version of the Starcounter binaries). 
```

> The easiest way to assure compatibility is use the same version of tooling and your application.

Reviewer 2: ```suggestion
The easiest way to assure compatibility is to use the same version of the tool as the version you use for your application.
```

> When a database fail to start, and you suspect that a versioning mismatch may be the problem, you can check the version used. The procedure is similar for both Starcounter tooling and any application referencing `Starcounter.Database`.

Reviewer: ```suggestion
When a database fails to start, and you suspect that a versioning mismatch may be the problem, you can check the version used. The procedure is similar for both Starcounter tooling and any application referencing `Starcounter.Database`.
```
